ratik/src/intTest/java/org/opencadc/ratik/InventoryValidatorTest.java

```diff
@@ -0,0 +1,560 @@
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.ratik;
+
+import ca.nrc.cadc.util.Log4jInit;
+import java.net.URI;
+import java.util.Calendar;
+import java.util.Set;
+import java.util.UUID;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.opencadc.inventory.Artifact;
+import org.opencadc.inventory.DeletedArtifactEvent;
+import org.opencadc.inventory.DeletedStorageLocationEvent;
+import org.opencadc.inventory.SiteLocation;
+import org.opencadc.inventory.StorageSite;
+
+public class InventoryValidatorTest {
+    private static final Logger log = Logger.getLogger(InventoryValidatorTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.ratik", Level.INFO);
+        Log4jInit.setLevel("org.opencadc.inventory", Level.INFO);
+        Log4jInit.setLevel("org.opencadc.inventory.db", Level.INFO);
+        Log4jInit.setLevel("ca.nrc.cadc.db", Level.INFO);
+    }
+
+    private final InventoryEnvironment localEnvironment = new InventoryEnvironment();
+    private final LuskanEnvironment remoteEnvironment = new LuskanEnvironment();
+
+    public InventoryValidatorTest() throws Exception {
+
+    }
+
+    @Before
+    public void beforeTest() throws Exception {
+        this.localEnvironment.cleanTestEnvironment();
+        this.remoteEnvironment.cleanTestEnvironment();
+    }
+
+    /**
+     * discrepancy: none
+     * before: Artifact in L & R
+     * after: Artifact in L & R
+     */
+    @Test
+    public void noDiscrepancy() throws Exception {
+        Calendar calendar = Calendar.getInstance();
+        Artifact artifact = new Artifact(URI.create("cadc:INTTEST/one.ext"), TestUtil.getRandomMD5(), calendar.getTime(), 1024L);
+        this.localEnvironment.artifactDAO.put(artifact);
+        this.remoteEnvironment.artifactDAO.put(artifact);
+
+        InventoryValidator testSubject = new InventoryValidator(this.localEnvironment.daoConfig, TestUtil.LUSKAN_URI, artifact.getBucket());
+        testSubject.run();
+
```

Comment:
for the test run, use of the uriBucket from the stored artifact would work but it should really be null to not complicate this test -- this also applicable to every other test.

**always use ArtifactDAO.get(UUID) (here and in other test methods) unless you really mean otherwise**. After a validate, the remote and local artifact should be identical, so the primary assert should be
```
Assert.assertEquals("metaChecksum", artifact.getMetaChecksum(), localArtifact.getMetaChecksum());
```
I should point out that ArtifactDAO.put(Artifact) does - as a side effect - assign metaChecksum and lastModified to the argument artifact. Putting the same one twice (lines 120-121) should obviously be fine and leave the artifact with the correct values (from the second put). It is intended to be done as a side effect so the values are available to the caller (eg for logging if desired) but it is subtle.

Reply:
Using null for uriBucket in the InventoryValidator constructor, and using the Artifact id for a get.